src/test/java/com/exactpro/th2/dataservice/healer/HealerTest.java

```diff
@@ -145,37 +156,27 @@ public void healedCorrectly() throws IOException {
                 .build();
 
         EventDataRequest request = EventDataRequest.newBuilder()
-                .setId(crawlerInfo.getId())
+                .setId(CRAWLER_INFO.getId())
                 .addEventData(parentEvent)
                 .addEventData(childEvent)
                 .addEventData(grandchildEvent)
                 .build();
 
-        healer.crawlerConnect(crawlerInfo, dataServiceResponseObserver);
+        blockingStub.crawlerConnect(CRAWLER_INFO);
+        blockingStub.sendEvent(request);
 
-        healer.sendEvent(request, eventResponseObserver);
-
-        verify(storageMock).updateEventStatus(events.get(0), false);
-        verify(storageMock).updateEventStatus(events.get(1), false);
-        verify(eventResponseObserver).onCompleted();
+        verify(STORAGE_MOCK).updateEventStatus(events.get(0), false);
+        verify(STORAGE_MOCK).updateEventStatus(events.get(1), false);
     }
 
     @Test
     public void crawlerUnknown() {
-        healer.sendEvent(
-                EventDataRequest.newBuilder()
-                        .setId(crawlerInfo.getId())
-                        .addEventData(EventData.getDefaultInstance())
-                        .build(), eventResponseObserver
-        );
-
-        EventResponse response = EventResponse.newBuilder()
-                .setStatus(Status.newBuilder().setHandshakeRequired(true))
-                .build();
+        EventResponse response = blockingStub.sendEvent(EventDataRequest.newBuilder()
+                .setId(CRAWLER_INFO.getId())
+                .addEventData(EventData.getDefaultInstance())
+                .build());
 
-        verify(eventResponseObserver).onNext(eq(response));
-        verify(eventResponseObserver).onCompleted();
-        verifyNoMoreInteractions(eventResponseObserver);
+        assertTrue(response.getStatus().getHandshakeRequired());
```

Comment:
Please, add the lambda to print the actual response in case the assertion is failed